src/osp/tasks/top_utils.h

```diff
@@ -0,0 +1,183 @@
+/**
+ * Open Space Program
+ * Copyright © 2019-2022 Open Space Program Project
+ *
+ * MIT License
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+#pragma once
+
+#include "tasks.h"
+#include "top_worker.h"
+
+#include <entt/core/any.hpp>
+
+#include <cassert>
+#include <type_traits>
+#include <utility>
+#include <vector>
+
+namespace osp
+{
+
+[[nodiscard]] inline TopDataId top_reserve(
+        ArrayView<entt::any> const topData, TopDataId current = 0)
+{
+    while ( bool(topData[current]) && (current < topData.size()) )
+    {
+        ++current;
```

Comment:
auto const& it = std::ranges::find(topData, true);
current = std::distance(std::begin(topData), it);

Reply:
Well this caused an unexpected segfault. For a comparison between entt::any and bool, the boolean is used to initialize a new entt::any with the bool. I may bug skypjack about this.